src/widgets/BadgeSet.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StyleSheet, View, TouchableHighlight, Text } from 'react-native';
+import { Badge } from 'react-native-elements';
+import { SUSSOL_ORANGE } from '../globalStyles';
+import Popover from 'react-native-popover-view';
+
+export class BadgeSet extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isVisible: false,
+      buttonRect: {},
+    };
+  }
+
+  showPopover() {
+    this.refs.button.measure((ox, oy, width, height, px, py) => {
+      this.setState({
+        isVisible: true,
+        buttonRect: { x: px, y: py, width: width, height: height },
+      });
+    });
+  }
+
+  closePopover() {
+    this.setState({ isVisible: false });
+  }
+
+  render() {
+    const {
+      MainElement,
+      finalizeValue,
+      mainWrapper,
+      popPlacement,
+    } = this.props;
+
+    const finalizeTxt = (finalizeValue > 99) ? '99+' : finalizeValue;
+    return (
+      <View style={{ position: 'relative' }}>
+      {MainElement}
+      {(finalizeValue !== 0) && (
+        <View style={[localStyles.mainWrapper, mainWrapper]}>
+          <TouchableHighlight ref="button" style={localStyles.touchPlaceholder} onPress={() => this.showPopover()}>
+            <Badge
+              value={finalizeTxt}
+              wrapperStyle={[localStyles.badgeWrapper, localStyles.finalize]}
+              containerStyle={localStyles.finalizeContainer}
+              textStyle={localStyles.badgeFont}
+            />
+          </TouchableHighlight>
+          <Popover
+            isVisible={this.state.isVisible}
+            fromRect={this.state.buttonRect}
+            onClose={() => this.closePopover()}
+            popoverStyle={{ padding: 10, backgroundColor: SUSSOL_ORANGE }}
+            arrowStyle={{ backgroundColor: SUSSOL_ORANGE }}
+            showBackground={false}
+            placement={popPlacement}
+          >
+            <Text style={{ color: '#FFF', fontSize: 10 }}>{finalizeTxt} Not Finalized</Text>
+          </Popover>
+        </View>
+      )}
+      </View>
+    );
+  }
+}
+
+BadgeSet.propTypes = {
+  MainElement: PropTypes.element.isRequired,
+  finalizeValue: PropTypes.number,
```

Comment:
And other references to finalise such as in the localStyles, `finalizeContainer` could be `valueContainer` instead.

Reply:
Last note - in Sussol's codebases we've tended towards English as we do in NZ/UK, where finalise is spelt with an 's' rather than a 'z'. Please search your branch and update all.

Reply:
> I think to make badge more generic, this prop should be just be `value`

@Chris-Petty The idea with the values was there would be many badges associated with a single button, say `finalise` or `new`. Expecting there to be 2-3 values as prop and BadgeSet would wrap and group them together.